### ISSUE TITLE
Bigtable emulator inconsistent with real bigtable when filtering only by rowkey

### ISSUE THREAD

**Issue by mtitolo** [googleapis/google-cloud-go#1355]
TL;DR Bigtable instances do not support reading rows with only a row key regex filter. The emulator supports this, which is very confusing. (The docs also say this is possible, but that's a different issue)

I have been able to reproduce this issue with both the Python and Ruby Bigtable clients. I've added to the python hello world sample to reproduce this issue: https://github.com/mtitolo/python-docs-samples/blob/d30e39cd4f3dddff7704706b5cc6d5534df5c190/bigtable/hello/main.py

Run once with `export BIGTABLE_EMULATOR_HOST=127.0.0.1:8086` and once with `unset BIGTABLE_EMULATOR_HOST` against a real project/instance.

## Client

Bigtable Emulator

## Describe Your Environment

macOS 10.14

## Expected Behavior

I expect the local emulator to respond to requests in the same way that a real instance would. This is the output from my hello world script for a real bigtable instance:

```
Creating the test table.
Creating column family cf1 with Max Version GC rule...
Writing some greetings to the table.
Getting a single greeting by row key.
Hello World!
Scanning for all greetings:
Hello World!
Hello Cloud Bigtable!
Hello Python!
Scanning for greetings by RowKeyRegex:
Deleting the test table.
```

There is an empty set returned when only reading rows by a row key regex filter, so no logs are printed between "Scanning for greetings by RowKeyRegex" and "Deleting the test table."

## Actual Behavior

Using only a row key regex filter does work locally, but it shouldn't. Here's the output from my hello world script talking to the emulator:
```
Creating the test table.
Creating column family cf1 with Max Version GC rule...
Writing some greetings to the table.
Getting a single greeting by row key.
Hello World!
Scanning for all greetings:
Hello World!
Hello Cloud Bigtable!
Hello Python!
Scanning for greetings by RowKeyRegex:
Hello World!
Hello Cloud Bigtable!
Hello Python!
Deleting the test table.
```

**Comment by odeke-em** [comment]
Thank you for this report @mtitolo and welcome to Google Cloud Go!

So in your bug report you report that 
>TL;DR Bigtable instances do not support reading rows with only a row key regex filter. The emulator supports this, which is very confusing.

And as per your source code in https://github.com/mtitolo/python-docs-samples/blob/d30e39cd4f3dddff7704706b5cc6d5534df5c190/bigtable/hello/main.py#L91
<img width="680" alt="Screen Shot 2019-03-20 at 1 27 24 PM" src="https://user-images.githubusercontent.com/4898263/54713307-ea5dfa00-4b13-11e9-8c61-0ef762f92e03.png">

However, I can't seem to reproduce your problem with your row_key_filter in Python at least.
### On live Bigtable instance
```shell
for ((i=0;i<10;i++))
do
   sleep 5
   python repro.py sandbox googleapis-tests > "run$i.txt"
done
```

producing each file with
```shell
Creating the Hello-Bigtable table.
Creating column family cf1 with Max Version GC rule...
Table Hello-Bigtable already exists.
Writing some greetings to the table.
Getting a single greeting by row key.
Hello World!
Scanning for all greetings:
Hello World!
Hello Cloud Bigtable!
Hello Python!
Scanning for greetings by RowKeyRegex:
```

and all those files are the same
```shell
$ md5 run*
MD5 (run0.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run1.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run2.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run3.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run4.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run5.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run6.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run7.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run8.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (run9.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
```

### On emulator
```shell
for ((i=0;i<10;i++))
do
   sleep 5
   BIGTABLE_EMULATOR_HOST=127.0.0.1:8086 python repro.py sandbox googleapis-tests > "run$i.txt"
done
```

produces each file with
```shell
Creating the Hello-Bigtable table.
Creating column family cf1 with Max Version GC rule...
Writing some greetings to the table.
Getting a single greeting by row key.
Hello World!
Scanning for all greetings:
Hello World!
Hello Cloud Bigtable!
Hello Python!
Scanning for greetings by RowKeyRegex:
```

and all those files are the same as when running with the live instance
```shell
$ md5 erun*
MD5 (erun0.txt) = fa730262de36fcc5cf22b4450e544b77
MD5 (erun1.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun2.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun3.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun4.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun5.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun6.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun7.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun8.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
MD5 (erun9.txt) = f5c61c9fb41577c080cff7b14bbbd1c6
```

*I didn't delete the test table for the sake of speeding things up*

Could you perhaps please try again? I am running an emulator at tip/master 942825909afd4837bd761b032cfb269ec165c3d3

**Comment by jadekler** [comment]
We've asked some clarifying questions and not heard back from you in a while. Closing this for now - please ping back if you'd like to continue the conversation.
